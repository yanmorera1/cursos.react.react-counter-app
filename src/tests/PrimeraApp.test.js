import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";

describe('Tests in <PrimeraApp>', () => {
    // test('should show message "Hola mundito"', () => {
    //     const saludo = "Hola mundito";
    //     const {getByText} = render(<PrimeraApp saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })
    test('should show <PrimeraApp/> successfully', () => {
        const saludo = "Hola Yan";
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();
    })

    test('should show subtitle sending by props', () => {
        const saludo = "Hola Yan";
        const subtitulo = "Rikito";
        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subtitulo}
            />
        )

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subtitulo);
    })
    
    
})
