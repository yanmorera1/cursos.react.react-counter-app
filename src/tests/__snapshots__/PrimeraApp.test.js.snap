// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <PrimeraApp> should show <PrimeraApp/> successfully 1`] = `
<Fragment>
  <h1>
    Hola Yan
    !!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
